code/modules/mob/mob.dm

```diff
@@ -1331,6 +1331,29 @@ var/list/slot_equipment_priority = list( \
 /mob/proc/is_literate()
 	return FALSE
 
+/mob/proc/faction_check_mob(mob/target, exact_match)
+	if(exact_match) //if we need an exact match, we need to do some bullfuckery.
+		var/list/faction_src = faction.Copy()
+		var/list/faction_target = target.faction.Copy()
+		if(!("\ref[src]" in faction_target)) //if they don't have our ref faction, remove it from our factions list.
```

Comment:
This *might* need to be switched to the `UID()` system, leaving a note here in case this becomes a bug somewhere down the line

Reply:
Does the "in" check still work if we use UID's here instead?